### PR TITLE
[IOPAY-405] update apim policy for new checkout APIs

### PR DESCRIPTION
**Pull request by infantesimone**
### List of changes

- enable checkout backend in all envs;
- merge `apim_checkout_payment_activations_api_v1` in  `apim_checkout_payments_api_v1`;
- add new policy for `getPaymentInfo` API in order to check google reCAPTCHA token in `recaptchaResponse` query param.
- add new policy for `CDInfoWisp` API in order to check node IP source;
- add new policy for `getActivationStaus` in order to retrive `idPayment`  from the `pagopa-proxy` in PROD-pagoPA or in PROD-IO ([more details](https://pagopa.atlassian.net/wiki/spaces/I/pages/440632330/RFC+-+porting+pagopa-proxy+in+pagoPA+draft#Opzione-2-Default));

### Motivation and context

The goal of this PR is to set the checkout api with the new `pagopa-proxy` in PROD-pagoPA. Some ad-hoc policies have been add to manage a transient in which the check of the _idPayment_ will be performed by calling the `pagopa-proxy` in PROD-IO
### Type of changes

- [x] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [x] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

**Review by giovanniberti — DISMISSED** [review]

File: src/env/dev/terraform.tfvars [hunk 1]
```diff
@@ -256,3 +256,4 @@ pagopa_proxy_redis_sku_name = "Basic"
 pagopa_proxy_redis_family   = "C"
 pagopa_proxy_tier           = "Standard"
 pagopa_proxy_size           = "S1"
+nodo_ip_filter              = ""
```
Comment (pasqualespica) [inline src/env/dev/terraform.tfvars:259]: why is empty ? and prod no t defined ?

File: src/api/checkout/checkout_payments/v1/_idpayment_check.xml.tpl [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      <!-- Check idPayment in IO env - END -->
+    </inbound>
+    <outbound>
+      <base />
+      <choose>
+        <when condition="@(context.Response.StatusCode == 404)">
+          <!-- Check idPayment in pagoPA env- START -->
+          <set-variable name="codice_contesto_pagamento" value="@(context.Request.MatchedParameters["codice_contesto_pagamento"])" />
+          <send-request ignore-error="true" timeout="10" response-variable-name="response" mode="new">
+            <set-url>$"{{pagopa-appservice-proxy-url}}/api/v1/payment-activations/${context.Variables["codice_contesto_pagamento"]}"</set-url>
```
Comment (pasqualespica) [inline src/api/checkout/checkout_payments/v1/_idpayment_check.xml.tpl:18]: `pagopa-appservice-proxy-url` is set to io pagopa-proxy right ?

File: src/apim_checkout.tf [hunk 3]
```diff
@@ -3,7 +3,6 @@
 ##############
 
 module "apim_checkout_product" {
-  count  = var.checkout_enabled || var.pagopa_proxy_enabled ? 1 : 0
   source = "git::https://github.com/pagopa/azurerm.git//api_management_product?ref=v1.0.16"
```
Comment (pasqualespica) [inline src/apim_checkout.tf:6]: why remove this one ?

Comment (infantesimone) [reply src/apim_checkout.tf:6]: checkout will be enabled in all environments, so I started a refactoring to remove all `count checks`

File: src/api/checkout/checkout_payments/v1/_idpayment_check.xml.tpl (hunk 2 above)
Comment (infantesimone) [reply src/api/checkout/checkout_payments/v1/_idpayment_check.xml.tpl:18]: yes, is pagopa-proxy deployed in pagoPA

File: src/env/dev/terraform.tfvars (hunk 1 above)
Comment (infantesimone) [reply src/env/dev/terraform.tfvars:259]: good point, I don't know them yet 😁

File: src/apim_checkout.tf (hunk 3 above)
Comment (infantesimone) [reply src/apim_checkout.tf]: reverted this refactoring, I will do it in a dedicated PR

**Review by MarcoAGreco — APPROVED** [review]

**Review by giovanniberti — DISMISSED** [review]

**Pull request merged**